### DISCUSSION TITLE
Working but status unhealthy in docker

### DISCUSSION THREAD

**Discussion by Bodo-von-Greif** [open-webui/open-webui#5908] [General]
Hi all,

i run open-webui in an Ubuntu 22.04.3 LTS docker environment.
Everything is working properly.

Start command is:
`docker run -e no_proxy=host.docker.internal -e http_proxy=http://my-proxy.de:8080 -e https_proxy=http://my-proxy.de:8080 -d --pull=always -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

In Portainer i see it status unhealthy.
Failure count: 9255
**Last output: jq: error (at <stdin>:0): break**

---

systemctl status ollama
● ollama.service - Ollama Service
     Loaded: loaded (/etc/systemd/system/ollama.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/ollama.service.d
             └─override.conf
     Active: active (running) since Thu 2024-09-26 08:44:27 CEST; 1 week 1 day ago
   Main PID: 3706552 (ollama)
      Tasks: 25 (limit: 76981)
     Memory: 3.3G
        CPU: 1d 8h 44min 12.249s
     CGroup: /system.slice/ollama.service
             └─3706552 /usr/local/bin/ollama serve

more /etc/systemd/system/ollama.service.d/override.conf
[Service]
Environment="HTTP_PROXY=http://my-proxy.de:8080/"
Environment="http_proxy=http://my-proxy:8080/"
Environment="HTTPS_PROXY=http://my-proxy.de:8080/"
Environment="https_proxy=http://my-proxy.de:8080/"
Environment="NO_PROXY=localhost,.my-domain.de,127.0.0.1,host.docker.internal"
Environment="no_proxy=localhost,.my-domain.de,127.0.0.1,host.docker.internal"
Environment="OLLAMA_HOST=0.0.0.0:11434"

ollama version is 0.3.12

Any ideas how to get rid of this?
Thanks in advance,
Bodo

**Comment by danteblank** [comment]
You got to look inside Dockerfile, there's healthcheck included there
https://github.com/open-webui/open-webui/blob/main/Dockerfile#L168

And healthcheck is waiting $PORT variable to be set, and if not, it will go to default 8080.

**Comment by Bodo-von-Greif** [comment]
Hi,

i had to add/change (maybe too much):

docker run ... -e no_proxy=localhost ... -e PORT=3000 ... -p 3000:3000 ...

override.conf:
Environment="PORT=3000"
Environment="port=3000"

Thanks a lot!!

PS: Problem: now i don't see my loaded models and coannot use them
Ollama know them:
ollama list
NAME                ID              SIZE      MODIFIED
llama2:latest       78e26419b446    3.8 GB    31 hours ago
llava:latest        8dd30f6b0cb1    4.7 GB    31 hours ago
mixtral:latest      a3b6bef0f836    26 GB     31 hours ago
codellama:latest    8fdf8f752f6e    3.8 GB    31 hours ago
llama3.3:latest     a6eb4748fd29    42 GB     31 hours ago

**Comment by Bodo-von-Greif** [comment]
When i do a helthcheck via prowser, i get:

Firefox: http://my-server.my-domain.de:3000/health
{
    "status": true
}

The health check of open-webui does:
jq -ne 'input.status == true'

So when i check with:
echo "{ "status": true }"|jq -ne 'input.status == true'
i get:
jq: error (at <stdin>:1): Invalid numeric literal at line 1, column 9

Where is my misunderstanding?

**Comment by danteblank** [comment]
You got to check it from inside docker.
docker exec -it openweb-ui bash

And then try original command.

If you want to use the one you typed:
echo '{"status": true}' | jq -ne 'input.status == true'

You got it wrong in your version.

**Reply by Bodo-von-Greif** [reply]
docker exec -it open-webui bash
root@b889e7cb12c0:/app/backend# echo '{"status": true}' | jq -ne 'input.status == true'
true

So this works, thanks a lot, but still no clue why it does not work

**Comment by danteblank** [comment]
Better use docker-compose instead of docker run.

**Reply by Bodo-von-Greif** [reply]
I already tried this but ended up with an new/empty volume

old run command
-  drwx-----x  8 root root      4096 Oct 24 15:43 open-webui/     

new docker-compose
 - drwx-----x  3 root root      4096 Jan 20 10:20 open-webui_open-webui/

so i copied the open-webui dirs to the new open-webui_open-webui directory

Start was ok and the data is correct.

But the unhealthy behaviour is unchanged.
After a few minutes of starting, it changes to unhealthy

docker-compose.yml
version: '3'
services:
  open-webui:
    image: ghcr.io/open-webui/open-webui:main
    environment:
      - no_proxy=host.docker.internal,0.0.0.0
      - http_proxy=http://xxx.de:8080
      - https_proxy=http://xxx:8080
    ports:
      - "3000:8080"
    volumes:
      - open-webui:/app/backend/data
    restart: always
    pull_policy: always
    extra_hosts:
      - "host.docker.internal:host-gateway"
volumes:
  open-webui:

docker exec -it open-webui-open-webui-1 bash
root@9034c902bc4d:/app/backend# echo '{"status": true}' | jq -ne 'input.status == true'
true

May this be a problem?
open-webui-1  | /usr/local/lib/python3.11/site-packages/aiohttp/connector.py:1048: RuntimeWarning: An HTTPS request is being sent through an HTTPS proxy. This support for TLS in TLS is known to be disabled in the stdlib asyncio (Python <3.11). This is why you'll probably see an error in the log below.
open-webui-1  |
open-webui-1  | It is possible to enable it via monkeypatching. For more details, see:
open-webui-1  | * https://bugs.python.org/issue37179
open-webui-1  | * https://github.com/python/cpython/pull/28073
open-webui-1  |
open-webui-1  | You can temporarily patch this as follows:
open-webui-1  | * https://docs.aiohttp.org/en/stable/client_advanced.html#proxy-support
open-webui-1  | * https://github.com/aio-libs/aiohttp/discussions/6044
open-webui-1  |
open-webui-1  |   _, proto = await self._create_proxy_connection(req, traces, timeout)
open-webui-1  | RuntimeWarning: Enable tracemalloc to get the object allocation traceback


Any further ideas appreciated

**Comment by Bodo-von-Greif** [comment]
Hi all,

i think i found the solution:
Had to add

```
healthcheck:
      test: "unset http_proxy;unset https_proxy;/usr/bin/curl --noproxy --fail --silent http://localhost:8080/health | jq -ne 'input.status == true'"
      interval: 10s
      timeout: 5s
      retries: 3
```

curl --noproxy does -not- ignore the proxy-settings!

Thanks for your time and attention!
Bodo
